script/test-race-conditions.spec.js

```diff
@@ -0,0 +1,378 @@
+ * arguments (if any) are the parameters for the method. Returns a function that
+ * will call the method with the arguments, as well as a callback.
+ *
+ * Use this when you expect that the method will fail.
+ *
+ * Example: methodFails("nameOfMethod", "param1", "param2", "param3");
+ *
+ * @return {Function}
+ */
+function methodFails() {
+    var args, method;
+
+    args = [].slice.call(arguments);
+    method = args.shift();
+
+    return (cb) => {
+        args.unshift((err) => {
+            if (err) {
+                cb(err);
+            } else {
+                cb(new Error("methodFails was called. No error was passed along."));
+            }
+        });
+        instance[method].apply(instance, args);
+    };
+}
+
+
+/**
+ * Sets up methods for functional tests. Accepts any number of arguments. The
+ * first argument is the name of the method to be called and the following
+ * arguments (if any) are the parameters for the method. Returns a function that
+ * will call the method with the argments as well as a callback.
+ *
+ * Use this when you expect that the method will succeed.
+ *
+ * Example: methodSucceeds("nameOfMethod", "param1", "param2", "param3");
+ *
+ * @return {Function}
+ */
+function methodSucceeds() {
+    var args, method;
+
+    args = [].slice.call(arguments);
+    method = args.shift();
+
+    return (cb) => {
+        args.unshift(cb);
+        instance[method].apply(instance, args);
+    };
+}
+
+
+/**
+ * Calls a list of methods one after another. This is used for testing that
+ * commands run as expected when they are not called in a reasoned, methodical
+ * order (i.e. one method calls the next in its callback).
+ *
+ * Example:
+ *   combine([
+ *       methodSucceeds("firstMethod"),
+ *       methodSucceeds("secondMethod"),
+ *       methodFails("thirdMethod")
+ *   ]);
+ *
+ * @param {array} list
+ * @return {Function}
+ */
+function combine(list) {
+    var remaining;
+
+    remaining = list.length;
+
+    return (testComplete) => {
+        /**
+         * The callback for each individual function in the list. Checks that
+         * there is another function in the list and breaks the flow if there
+         * isn't or if an error has occured.
+         *
+         * @param {Error} err
+         */
+        function singleTestDone(err) {
+            if (!remaining) {
+                return;
+            }
+
+            if (err) {
+                remaining = 0;
+                testComplete(err);
+            } else {
+                remaining -= 1;
+
+                if (!remaining) {
+                    testComplete();
+                }
+            }
+        }
+
+        list.forEach((fn) => {
+            fn(singleTestDone);
+        });
+    };
+}
+
+
+beforeEach(() => {
+    instance = new DDQ(config);
+    instance.on("error", (err) => {
+        done(err);
+    });
+    spyOn(instance, "close").andCallThrough();
```

Comment:
I believe `.andCallThrough()` is the default so you shouldn't need to include it for these.